base/intfuncs.jl

```diff
@@ -805,13 +805,23 @@ julia> bitstring(2.2)
 "0100000000000001100110011001100110011001100110011001100110011010"
 ```
 """
-function bitstring end
-
-bitstring(x::Union{Bool,Int8,UInt8})           = string(reinterpret(UInt8,x), pad = 8, base = 2)
-bitstring(x::Union{Int16,UInt16,Float16})      = string(reinterpret(UInt16,x), pad = 16, base = 2)
-bitstring(x::Union{Char,Int32,UInt32,Float32}) = string(reinterpret(UInt32,x), pad = 32, base = 2)
-bitstring(x::Union{Int64,UInt64,Float64})      = string(reinterpret(UInt64,x), pad = 64, base = 2)
-bitstring(x::Union{Int128,UInt128})            = string(reinterpret(UInt128,x), pad = 128, base = 2)
+function bitstring(x::T) where {T}
+    isprimitivetype(T) || throw(ArgumentError("$T not a primitive type"))
+    sz = sizeof(T) * 8
+    str = Base.StringVector(sz)
+    i = sz
+    @inbounds while i >= 4
+        b = UInt32(x isa UInt8 ? x : Base.trunc_int(UInt8, x))
```

Comment:
There is still an error here if `x` is of size 8, no?

```jl
julia> Base.trunc_int(UInt8, Int8(3))
ERROR: Trunc: output bitsize must be < input bitsize
   ```

Reply:
Yep, good catch. Just pushed a fix.